tree/30_generic_methods/sysctl_value.cf

```diff
@@ -0,0 +1,213 @@
+      "full_inner_class_prefix_variable_get"
+                              string => canonify("variable_string_from_command_sysctl_var_${getKey}_${sysctl_get_query}");
+      "inner_class_prefix_variable_get"
+                              string => string_head("${full_inner_class_prefix_variable_get}", "1000");
+
+      # we fetch on the getKey
+      "checkKey"              string => canonify("check${key}");
+      "full_inner_class_prefix_variable_check"
+                              string => canonify("variable_string_from_command_sysctl_var_${checkKey}_${sysctl_check_query}");
+      "inner_class_prefix_variable_check"
+                              string => string_head("${full_inner_class_prefix_variable_check}", "1000");
+
+
+      # reloading the configuration. In info we let it output its output
+    info|debug::
+      "sysctl_reload"         string => "${paths.sysctl} --system  # Reload value ${key} (expect ${value})";
+    !(info|debug)::
+      "sysctl_reload"         string => "${paths.sysctl} --system  > /dev/null # Reload value ${key} (expect ${value})";
+    any::
+      "full_inner_class_prefix_sysctl_reload"
+                              string => canonify("command_execution_${sysctl_reload}");
+      "inner_class_prefix_sysctl_reload"
+                              string => string_head("${full_inner_class_prefix_sysctl_reload}", "1000");
+
+  defaults:
+      "option"                string => "default", if_match_regex => "";
+
+  classes:
+      # Sanity check : need sysctl.d folder, need redhat7 or more, debian 7 or more, sles12 or more, and 
+      "sysctl_d_exists"   expression => isdir("/etc/sysctl.d");
+      "can_edit"          expression => "sysctl_d_exists.!(debian_3|debian_4|debian_5|debian_6|redhat_3|redhat_4|redhat_5|redhat_6|sles_9|sles_10|sles_11|ubuntu_12|ubuntu_10|aix|solaris)";
+
+      "should_report"     expression => "${report_data.should_report}";
+
+      "check_smaller"     expression => strcmp("${option}", "min");
+      "check_larger"      expression => strcmp("${option}", "max");
+
+      "check_equality"           not => "check_larger|check_smaller";
+
+    pass1::
+      "correct_value"     expression => strcmp("${sysctl_var.${getKey}}", "${value}");
+
+      "correct_value"     expression => isgreaterthan("${sysctl_var.${getKey}}", "${value}"),
+                                  if => "check_smaller";
+
+      "correct_value"     expression => islessthan("${sysctl_var.${getKey}}", "${value}"),
+                                  if => "check_larger";
+
+    pass2::
+      # check post modification
+      "correct_value_post_modif"
+                          expression => strcmp("${sysctl_var.${checkKey}}", "${value}"),
+                                  if => "${inner_class_prefix_sysctl_reload}_repaired";
+
+    any::
+      "pass3" expression => "pass2";
+      "pass2" expression => "pass1";
+      "pass1" expression => "any";
+
+  methods:
+      "disable_reporting_${class_prefix}"
```

Comment:
should not we use full_class_prefix everywhere?

Reply:
YES

Reply:
bad copy/paste